conans/client/generators/virtualenv.py

```diff
@@ -1,9 +1,106 @@
 import os
+import platform
+import textwrap
+
+from jinja2 import Template
 
 from conans.client.tools.oss import OSInfo
 from conans.model import Generator
 
 
+sh_activate_tpl = Template(textwrap.dedent("""
+    #!/usr/bin/env bash
+    # DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+    # DIR="$( cd "$( dirname "$_" )" >/dev/null 2>&1 && pwd )"
+
+    {%- for it in modified_vars %}
+    export CONAN_OLD_{{it}}="${{it}}"
+    {%- endfor %}
+
+    while read line; do
```

Comment:
Maybe it's too late to ask, but it seems a bit inconsistent to me:
Remembering old variables is done here explicitly, while setting new vars are read from the file.

Besides, I can't find quickly where concatenation with the current environment happens...

Major problem I have with virtual environment generator using `os.environ` is that it captures variables at the moment of calling `conan install`. So if I (happen to) run it from an activated environment (probably from another project or with another settings / options) the activate and deactivate scripts will be a total mess.

Reply:
I'm moving the conversation here: https://github.com/conan-io/conan/pull/5989#issuecomment-562142696